### PR TITLE
Updated the module to use seal-secrets submodule instead of kubeseal download and fixed the issue faced during CNTK BOM install

### LINKED ISSUE #11: The README has not been updated

### LINKED ISSUE #14: Sonarqube install on Azure is failing with the following error while creating replica set: Sonarqube: forbidden: unable to validate against any security context constraint
Complete error : `Error creating: pods "azure-cntk-2-services-sonarqube-sonarqube-746678b75b-" is forbidden: unable to validate against any security context constraint: [provider "anyuid": Forbidden: not usable by user or serviceaccount, provider "gitops-openldap-openldap-anyuid": Forbidden: not usable by user or serviceaccount, provider "pipelines-scc": Forbidden: not usable by user or serviceaccount, provider "sealed-secrets-sealed-secrets-anyuid": Forbidden: not usable by user or serviceaccount, provider "tools-artifactory-artifactory-anyuid": Forbidden: not usable by user or serviceaccount, provider "tools-sonarqube-anyuid": Forbidden: not usable by user or serviceaccount, provider "turbonomic-t8c-operator-anyuid": Forbidden: not usable by user or serviceaccount, provider restricted: .spec.securityContext.fsGroup: Invalid value: []int64{1000}: 1000 is not an allowed group, spec.initContainers[0].securityContext.privileged: Invalid value: true: Privileged containers are not allowed, spec.containers[0].securityContext.runAsUser: Invalid value: 1000: must be in the ranges: [1000920000, 1000929999], provider "nonroot": Forbidden: not usable by user or serviceaccount, provider "hostmount-anyuid": Forbidden: not usable by user or serviceaccount, provider "machine-api-termination-handler": Forbidden: not usable by user or serviceaccount, provider "hostnetwork": Forbidden: not usable by user or serviceaccount, provider "hostaccess": Forbidden: not usable by user or serviceaccount, provider "node-exporter": Forbidden: not usable by user or serviceaccount, provider "cntk-ocp-userspaces-daemonset-privileged": Forbidden: not usable by user or serviceaccount, provider "gitops-openldap-openldap-privileged": Forbidden: not usable by user or serviceaccount, provider "privileged": Forbidden: not usable by user or serviceaccount, provider "privileged-genevalogging": Forbidden: not usable by user or serviceaccount, provider "tools-artifactory-artifactory-privileged": Forbidden: not usable by user or serviceaccount, provider "tools-sonarqube-privileged": Forbidden: not usable by user or serviceaccount]`

### LINKED ISSUE #17: The module should use the seal-secrets submodule instead of downloading its own version of kubeseal
Should follow the pattern established here - https://github.com/cloud-native-toolkit/terraform-gitops-cp-bas

### PR DESCRIPTION
**Pull request by RamyaRaghuveera**
closes #17, #14, and #11 
Signed-off-by: Ramya Raghuveera <ramragh1@in.ibm.com>

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -158,7 +159,20 @@ module setup_clis {
 resource null_resource setup_chart {
   provisioner "local-exec" {
     command = "${path.module}/scripts/create-yaml.sh '${local.yaml_dir}' '${local.service_url}' '${var.namespace}' '${local.values_file}'"
+    environment = {
+      VALUES_CONTENT = yamlencode(local.values_content)
+      VALUES_SERVER_CONTENT = yamlencode(local.values_server_content)
+      TMP_DIR = local.tmp_dir
+      KUBESEAL_CERT = var.kubeseal_cert
```
Comment (seansund) [inline main.tf]: KUBESEAL_CERT can be removed
SECRET_DIR = local.secret_dir

File: scripts/create-secrets.sh [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+  #KUBESEAL=$(command -v kubeseal | command -v ./bin/kubeseal)
+  #if [[ -z "${KUBESEAL}" ]]; then
+    #BIN_DIR=$(cd ./bin; pwd -P)
+    #mkdir -p "${BIN_DIR}" && curl -Lo "${BIN_DIR}/kubeseal" https://github.com/bitnami-labs/sealed-secrets/releases/download/v0.16.0/kubeseal-linux-amd64
+    #chmod +x "${BIN_DIR}/kubeseal"
+    #KUBESEAL="${BIN_DIR}/kubeseal"
+  #fi
+
+  echo "Kubeseal cert"
```
Comment (seansund) [inline scripts/create-secrets.sh]: This does not belong here. The Sealed Secrets submodule handles this. Should follow the pattern established here - https://github.com/cloud-native-toolkit/terraform-gitops-cp-bas

File: scripts/create-yaml.sh [hunk 3]
```diff
@@ -53,21 +55,22 @@ stringData:
...
+    #mkdir -p "${BIN_DIR}" && curl -Lo "${BIN_DIR}/kubeseal" https://github.com/bitnami-labs/sealed-secrets/releases/download/v0.16.0/kubeseal-linux-amd64
+    #chmod +x "${BIN_DIR}/kubeseal"
+    #KUBESEAL="${BIN_DIR}/kubeseal"
+  #fi
 
   echo "Kubeseal cert"
   cat "${KUBESEAL_CERT_FILE}"
-
+  KUBESEAL=$(command -v kubeseal | command -v ./bin/kubeseal)
   ${KUBESEAL} --cert "${KUBESEAL_CERT_FILE}" --format yaml < "${SECRET_FILE}" > "${TEMPLATE_DIR}/sonarqube-access.yaml"
```
Comment (seansund) [inline scripts/create-yaml.sh]: This isn't necessary here

**Review by seansund — CHANGES_REQUESTED** [review]
Needs several updates. Follow the pattern established here for create yaml, create secret, seal secret - https://github.com/cloud-native-toolkit/terraform-gitops-cp-bas

**Comment by RamyaRaghuveera** [comment]
@seansund I have merged the changes of #15 into this.

File: main.tf [hunk 4]
```diff
@@ -26,7 +27,7 @@ locals {
       enabled = false
     }
     serviceAccount = {
-      create = false
+      create = true
```
Comment (seansund) [inline main.tf:30]: The service account should be created with a submodule

File: test/stages/stage1-namespace.tf [hunk 5]
```diff
@@ -4,7 +4,7 @@ module "gitops_namespace" {
   gitops_config = module.gitops.gitops_config
   git_credentials = module.gitops.git_credentials
   name = var.namespace
-  server_name              = module.gitops.server_name
```
Comment (seansund) [inline test/stages/stage1-namespace.tf:7]: Server name should be provided to the test. Why was this removed?

**Review by seansund — CHANGES_REQUESTED** [review]
two more changes

File: main.tf (hunk 4 above)
Comment (seansund) [reply main.tf:30]: Based on comments from @RamyaRaghuveera , this value is required to use a different service account other than default. It causes a warning in ArgoCD but we can revisit this when we upgrade the helm chart for SonarQube

**Review by seansund — APPROVED** [review]
We will need to revisit the service account creation later but these changes look good otherwise

File: main.tf (hunk 4 above)
Comment (RamyaRaghuveera) [reply main.tf:30]: Adding the deployment.yaml snippet with create flag for future reference:
{{- if .Values.serviceAccount.create }}
    {{- if .Values.serviceAccount.name }}
      serviceAccountName: {{ .Values.serviceAccount.name }}
    {{- else }}
      serviceAccountName: {{ include "sonarqube.fullname" . }}
    {{- end }}
    {{- end }}
